### PR TITLE
fix(retry strategy): Shared appId host usage, retry on first host after 1 minute

### PR DESCRIPTION
**Pull request by vvo**
This commit adds another layer of HA by remembering the last known
work host for one minute.

The cache keys are either:
- domain + appId + browser => Browser localStorage
- process/pageLoad + appId => Node.js, Browser fallback

This is still a bit raw I think, have some review.

### CODE REVIEW

**Comment by bobylito** [comment]
Is going back to the first host (the DSN in the case of Algolia default values) make sense when it comes to hosts provided by our users?

**Comment by rayrutjes** [comment]
Shouldn't we retry after 5 minutes instead of 1 minute?

File: src/AlgoliaSearchCore.js [hunk 1]
```diff
@@ -3,9 +3,11 @@ module.exports = AlgoliaSearchCore;
 var errors = require('./errors');
 var exitPromise = require('./exitPromise.js');
 var IndexCore = require('./IndexCore.js');
+var store = require('./store.js');
 
 // We will always put the API KEY in the JSON body in case of too long API KEY
 var MAX_API_KEY_LENGTH = 500;
+var RESET_TO_FIRST_HOST_TIMER = 60 * 60 * 1000;
```
Comment (bobylito) [inline src/AlgoliaSearchCore.js]: Isn't that an hour?

Comment (redox) [inline src/AlgoliaSearchCore.js]: Good catch 👍

Comment (vvo) [inline src/AlgoliaSearchCore.js]: Thx, updated to 5min

**Comment by vvo** [comment]
> Shouldn't we retry after 5 minutes instead of 1 hour?

updated

**Comment by vvo** [comment]
Added a unit test around retrying the first host

**Comment by vvo** [comment]
- unit tests :white_check_mark: Chrome latest
- unit tests :white_check_mark: IE9
- unit tests :white_check_mark: IE10
- unit tests :white_check_mark: IE11

**Pull request merged**